### ISSUE TITLE
Output Channel - Highlight

### ISSUE THREAD

**Issue by YongdongHe** [Microsoft/vscode#37666]
![image](https://user-images.githubusercontent.com/9399699/32415230-34dc7ba6-c271-11e7-81e4-f8ff3447769d.png)

I used a language id to create a output channel with this way:
https://github.com/Microsoft/vscode/issues/13502

It's still take effect on debug mode , but does not take effect after the package.

**Comment by YongdongHe** [comment]
My extension code as below:
https://github.com/YongdongHe/RapidView/tree/master/rapidstudio_vscode_extension/rapidstudio

**Comment by isidorn** [comment]
@YongdongHe Can you please clarify what is the issue here.
What is the expected behavior, what behavior are you seeing?
Can you also reproduce this with latest vscode insiders https://code.visualstudio.com/insiders/

**Comment by YongdongHe** [comment]
Thanks for isidorn's response.
This problem is repeated every time with the latest vscode .
What I expected is  highlighting the output in outputchannel from my extension by creating a language + grammar and associating it with text/x-code-output mimetype.To do this, the user of extension can know which of these messages in outputchanel is the error or success message.
I've succeeded in debugging mode. But once I put it on the market or use the vsce command to package it and then install it from the market or vsix, I do not see the highlight。
As the picture shows, I install my extension from vsix for my vscode, and here is no highlight.
But if I press F5 to debug my extension, I can see the highlight.
Is there any difference between debugging mode and normal mode reading configuration file here? This does not seem reasonable.

**Comment by isidorn** [comment]
@YongdongHe please try reproducign this with latest vscode insiders https://code.visualstudio.com/insiders/
Not with latest stable vscode.

@joaomoreno is there some difference between a vsix packaged extension and the one being debugged?

**Comment by YongdongHe** [comment]
Yes, I can reproduce it with lastest vscode insiders.

**Comment by sandy081** [comment]
@isidorn / @joaomoreno Can you help in reproducing it to know the root cause?

**Comment by bpasero** [comment]
@joaomoreno well in terms of ordering, the debugged extensions always comes first in terms of what it registers, so it will overwrite any other extension that contributes the same language mode. 

Could it be that here we just have 2 extensions installed that both contribute to the output channel and we just pick the other one to win?

**Comment by YongdongHe** [comment]
That sounds good if we can pick one of extensions to win.
After all, I do not know whether my extension will be overwritten by other extensions.
But this problem does not seem to be caused by this, because I have a custom language Id called "rapid-log", it should not be covered by other extensions.
I'm trying to read the source code, but it's still a bit difficult to find the root cause.

**Comment by alexandrudima** [comment]
When reducing the extension to its bare necessities:
* `package.json`
```
{
    "publisher": "alex",
    "name": "37666",
    "version": "0.0.0",
    "engines": {
        "vscode": "^1.0.0"
    },
    "contributes": {
        "languages": [
            {
                "id": "rapid-log",
                "extensions": [
                    ".log"
                ],
                "mimetypes": [
                    "log",
                    "text/log",
                    "text/x-log",
                    "text/x-code-output",
                    "x-code-output"
                ]
            }
        ],
        "grammars": [
            {
                "language": "rapid-log",
                "scopeName": "source.log",
                "path": "./log.tmLanguage"
            }
        ]
    }
}
```
* `log.tmLanguage`: https://raw.githubusercontent.com/YongdongHe/RapidView/master/rapidstudio_vscode_extension/rapidstudio/src/syntaxes/log.tmLanguage

---

And placing the extension in my installation folder, I cannot reproduce. i.e. The coloring is applied. I have tried for both Insiders and running out of source. @YongdongHe my bet is that you are suffering from a **collision** of sorts. i.e. What other extensions do you have installed ? Most likely, one of them already targets the same mime type of `text/x-code-output`

**Comment by YongdongHe** [comment]
Thanks very much for your help.
I have checked all `package.json`  file  of extension installed. I'm sure there are no extension target  the mime type of `text/x-code-output`. And the insiders didn't install  any extension.So I don't think I am suffering from a collision of sorts. 

In this case, it may be because of the other settings in package.json affected?
Now that you can't reproduce it , I can only try to find the reason by myself.

Thank you anyway! And can you tell me the version of vsce you use?

**Comment by YongdongHe** [comment]
I find that the custom snippets in extension also only applied in debug mode.
I have uninstall all other extensions except TSLint.

**Comment by YongdongHe** [comment]
@isidorn @alexandrudima 
I have found the reason: 
I have put the log.tmLanguage  file under ./src folder, which leads to it can be found in the debug mode, but it will not be included when the .ts file were complied to .js file.

Becasue after the compiling, all .ts file were changed to .js file which were placed in the ./out folder. There is no ./src folder any more.

I think the document should indicate this.
